src/utils/DarCollectionUtils.js

```diff
@@ -0,0 +1,120 @@
+import { flow, isEmpty, map, join, filter, forEach, flatMap, toLower, sortBy, isNil } from 'lodash/fp';
+import {translateDataUseRestrictionsFromDataUseArray} from '../libs/dataUseTranslation';
+
+//Initial step, organizes raw data for further processing in later function/steps
+export const generatePreProcessedBucketData = async ({dars, datasets}) => {
+  const dataUses = [];
+  const buckets = {};
+  const datasetBucketMap = {};
```

Comment:
minor nit ... these const's are mutated. Might be better to use let here for clarity of intention.

Reply:
Actually on second thought I want to keep it as a const. The elements are being accessed by reference, which I can only guarantee with the Objects being a const. If I used a let then it's possible to replace the map/bucket with a new, blank object, which will have adverse effects downstream. 

The code doesn't allow this to happen, but should someone try to do so recklessly eslint will throw a compilation error with const compared to let.